### ISSUE TITLE
Method show() does not allow colorbar

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/GeoUtils#93]
rasterio's show method does not allow adding a color bar. Find a solution to this...

**Comment by jlandmann** [comment]
You can use imshow() in rasterio directly:

     import rasterio
    from matplotlib import pyplot
    src = rasterio.open("some.tif")
    pyplot.imshow(src.read(1), cmap='pink')
    pyplot.colorbar()
    pyplot.show()
![grafik](https://user-images.githubusercontent.com/10832560/109856951-634fb100-7c5a-11eb-9eee-aebc3b75e612.png)

**Comment by adehecq** [comment]
> You can use imshow() in rasterio directly:

Well but then you lose the georeferences... I can of course do it in matplotlib but I was wondering if there was a more direct way.

**Comment by rhugonnet** [comment]
same code than Johannes but with cartopy (>0.17) will get imshow() to handle the georeferences:
```python
import cartopy
import matplotlib.pyplot as plt
import rasterio as rio
src = rio.open('/home/atom/ongoing/work_worldwide/N65W019_final_err.tif')
fig = plt.figure()
ax = fig.add_subplot(1, 1, 1, projection=cartopy.crs.Robinson())
ax.gridlines(draw_labels=True, dms=True, x_inline=False, y_inline=False)
plt.imshow(src.read(1),cmap=plt.cm.Reds)
plt.colorbar()
plt.show()
```
![Screenshot from 2021-03-03 20-40-45](https://user-images.githubusercontent.com/28896516/109862522-d2300880-7c60-11eb-864e-8fd8e5ab5ceb.png)
Would that work?

**Comment by jlandmann** [comment]
Very cool! So just for the record, here's how it can work with rasterio:

      import rasterio
      from matplotlib import pyplot
      from matplotlib import colors, cm
      from rasterio.plot import show
      src = rasterio.open("some.tif")
      fig, ax = plt.subplots()
      cmap = pyplot.get_cmap('rainbow')
      data = src.read()
      show(data , transform=src.transform, ax=ax, cmap=cmap)
      fig.colorbar( cm.ScalarMappable(norm=colors.Normalize(vmin=np.nanmin(data), vmax=np.nanmax(data)), cmap=cmap), ax=ax)

![grafik](https://user-images.githubusercontent.com/10832560/109863005-669a6b00-7c61-11eb-902a-e1b8cfc1a6b9.png)

**Comment by adehecq** [comment]
Solved by #95
